R/evaluate-clusters.R

```diff
@@ -29,25 +37,33 @@
 calculate_silhouette <- function(
     x,
     cluster_df,
+    cluster_col = "cluster",
+    cell_id_col = "cell_id",
     pc_name = NULL) {
   x <- prepare_pc_matrix(x, pc_name)
 
-  expected_df_names <- c("cell_id", "cluster")
+  expected_df_names <- c(cell_id_col, cluster_col)
   stopifnot(
-    "Expected columns 'cell_id' and 'cluster' in the cluster_df." =
+    "Expected columns not present in cluster_df." =
       all(expected_df_names %in% colnames(cluster_df))
   )
 
+  cluster_col_symbol <- as.symbol(cluster_col)
+
   silhouette_df <- x |>
-    bluster::approxSilhouette(cluster_df$cluster) |>
+    bluster::approxSilhouette(cluster_df[[cluster_col]]) |>
     as.data.frame() |>
-    tibble::rownames_to_column("cell_id") |>
-    dplyr::rename("silhouette_width" = "width")
+    tibble::rownames_to_column(cell_id_col) |>
+    dplyr::rename(
+      "silhouette_width" = "width",
+      # ensure cluster column name matches the one provided
+      {{cluster_col_symbol}} := "cluster"
+    )
 
   # join with cluster_df in this direction, so that columns in
   # cluster_df come first
   silhouette_df <- cluster_df |>
-    dplyr::inner_join(silhouette_df, by = c("cell_id", "cluster"))
+    dplyr::inner_join(silhouette_df, by = c(cell_id_col, cluster_col))
```

Comment:
You may be able to avoid the rlang stuff by just setting the `by` statement here. Because the first table is the one with variable names, it's a bit strange. Leaving the original silhouette column names, this should work:

```
by = setNames(c("cell_id", "cluster"), c(cell_id_col, cluster_col)
```

Reply:
✅